yorkie/src/main/kotlin/dev/yorkie/document/crdt/TreeInfo.kt

```diff
@@ -8,23 +8,14 @@ import dev.yorkie.util.IndexTreeNode.Companion.DEFAULT_TEXT_TYPE
  * [TreeNode] represents the JSON representation of a node in the tree.
  * It is used to serialize and deserialize the tree.
  */
-internal data class TreeNode(
-    val type: String,
-    val children: List<TreeNode>? = null,
-    val value: String? = null,
-    val attributes: Map<String, String>? = null,
-) {
+internal sealed class TreeNode {
+    abstract val type: String
+    abstract val childNodes: List<TreeNode>?
+    abstract val value: String?
+    abstract val attributes: Map<String, String>?
```

Comment:
I think it can be refactored to have only `type` as abstract, with the other properties implemented in each subclass.

Reply:
done at [8392c71](https://github.com/yorkie-team/yorkie-android-sdk/pull/174/commits/8392c7143ce7ec0ec881596ee138eb57b0c272ae)